6_VendingMachine.cpp

```diff
@@ -0,0 +1,217 @@
+
+Sold out
+Not fully paid
+Not enough changes
+
+What is it? Simple vending machine
+Who's going to use it? Customer
+When will it be used? Anytime
+Where will it be used? Any place
+Why will it be used? Automatic sales of general use products
+How will it be used? Upon request by the customer, fetch prices, accept payment, render the product.
+
+
+2. E - EXAMPLES
+VendingMachine vm = new VendingMachine();		Create a new vending machine with unique id;
+vm.addStock(item, amount);						Add a certain amount of items into item list;
+vm.getRequest(request);							User send a request(with a selected item)
+vm.cancelRequest();								Cancel current user request;
+
+
+3. B - BRUTE-FORCE
+
+Create one class: VendingMachine, which contains all the state and behavior.
+
+4. O - OPTIMISE
+
+Create different classes for User requests, products, currency and Vending Machine
+
+5. W - WALK THROUGH
+
+Explained as comments in the code
+
+6. I - IMPLEMENT
+*/
+#include <iostream>
+#include <stdlib.h>
+using namespace std;
+//class for coins
+class currency
+{
+    public:
+        int coins_5;
+        int coins_10;
+    public:
+        currency()
+        {
+            coins_5=100;
+            coins_10=50;
+        }
+};
+int id=0;
+class Product
+{
+	int p_id;
+	float p_price;
+	public:
+        int p_count;
+        Product()
+        {
+            p_id=id++;
+            p_price=100;
+            p_count=10;
+        }
+
+        void increase_p_count()
+        {
+            p_count++;
+        }
+        void decrease_p_count()
+        {
+            p_count--;
+        }
+        float get_price()
+        {
+            return p_price;
+        }
+};
+
+class Vending_Machine : public currency
+{
+	float curr_balance;
+	int available_items[3];
+	currency* coin;
+	public:
+        Product* items_list[3];
+        Vending_Machine()
+        {
+            curr_balance=1000;
+            for (int i=0;i<3;i++)
+            {
+                items_list[i]= new Product();
+                available_items[i]=1;
+            }
+            coin=new currency();
+        }
+
+void add_coins(int cnt_5,int cnt_10)
+{
+    coins_10+=cnt_10;
+    coins_5+=cnt_5;
+}
+void pay_price(float bal)
+{
+    float total=0;
+    if(curr_balance>=bal)
+    {
+        while(total<=bal && coins_10>0)
+        {
+            total+=10;
+	      	coins_10-=1;
+        }
+        while(total<=bal && coins_5>0)
+        {
+	      	total+=5;
+	      	coins_5-=1;
+        }
+
+        if(total<bal)
+	      	cout <<"Insufficient change";
+    }
+    else
+     	cout <<"Insufficient change";
+}
+
+void update_VM(float amount,int item_id)
+{
+    float x=items_list[item_id]->get_price();
+    curr_balance+=x;
+    float bal=amount-x;
+    items_list[item_id]->decrease_p_count();
+    pay_price(bal);
+}
+};
+
+int main()
+{
+    Vending_Machine VM;
+	cout<< "Welcome!";
+	cout<<"\nChoose your option\n 1.Request the price of an item\n2. Select and buy items\n3. Cancel request";

```

Comment:
Is this how a vending machine usually works?  Typically you can select an item or add money in any order and different things happen based on the case